src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -616,130 +615,285 @@ .maxstack  5
+    IL_0079:  ldloca.s   V_3
+    IL_007b:  ldloca.s   V_4
+    IL_007d:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<int>.AwaitUnsafeOnCompleted<System.Runtime.CompilerServices.TaskAwaiter<int>, Test.<F>d__2>(ref System.Runtime.CompilerServices.TaskAwaiter<int>, ref Test.<F>d__2)""
+    IL_0082:  nop
+    IL_0083:  leave      IL_011c
+   >IL_0088:  ldarg.0
+    IL_0089:  ldfld      ""System.Runtime.CompilerServices.TaskAwaiter<int> Test.<F>d__2.<>u__1""
+    IL_008e:  stloc.3
+    IL_008f:  ldarg.0
+    IL_0090:  ldflda     ""System.Runtime.CompilerServices.TaskAwaiter<int> Test.<F>d__2.<>u__1""
+    IL_0095:  initobj    ""System.Runtime.CompilerServices.TaskAwaiter<int>""
+    IL_009b:  ldarg.0
+    IL_009c:  ldc.i4.m1
+    IL_009d:  dup
+    IL_009e:  stloc.0
+    IL_009f:  stfld      ""int Test.<F>d__2.<>1__state""
+    IL_00a4:  ldarg.0
+    IL_00a5:  ldloca.s   V_3
+    IL_00a7:  call       ""int System.Runtime.CompilerServices.TaskAwaiter<int>.GetResult()""
+    IL_00ac:  stfld      ""int Test.<F>d__2.<>s__3""
+    IL_00b1:  ldarg.0
+    IL_00b2:  ldarg.0
+    IL_00b3:  ldfld      ""int[] Test.<F>d__2.<>s__5""
+    IL_00b8:  ldc.i4.3
+    IL_00b9:  ldarg.0
+    IL_00ba:  ldfld      ""int Test.<F>d__2.<>s__2""
+    IL_00bf:  ldarg.0
+    IL_00c0:  ldfld      ""int Test.<F>d__2.<>s__3""
+    IL_00c5:  add
+    IL_00c6:  dup
+    IL_00c7:  stloc.2
+    IL_00c8:  stelem.i4
+    IL_00c9:  ldloc.2
+    IL_00ca:  stfld      ""int Test.<F>d__2.<>s__4""
+    IL_00cf:  ldarg.0
+    IL_00d0:  ldfld      ""int Test.<F>d__2.<>s__1""
+    IL_00d5:  ldarg.0
+    IL_00d6:  ldfld      ""int Test.<F>d__2.<>s__4""
+    IL_00db:  ldc.i4.4
+    IL_00dc:  call       ""int Test.H(int, int, int)""
+    IL_00e1:  pop
+    IL_00e2:  ldarg.0
+    IL_00e3:  ldnull
+    IL_00e4:  stfld      ""int[] Test.<F>d__2.<>s__5""
+   -IL_00e9:  ldc.i4.1
+    IL_00ea:  stloc.1
+    IL_00eb:  leave.s    IL_0107
+  }
+  catch System.Exception
+  {
+   ~IL_00ed:  stloc.s    V_5
+    IL_00ef:  ldarg.0
+    IL_00f0:  ldc.i4.s   -2
+    IL_00f2:  stfld      ""int Test.<F>d__2.<>1__state""
+    IL_00f7:  ldarg.0
+    IL_00f8:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder<int> Test.<F>d__2.<>t__builder""
+    IL_00fd:  ldloc.s    V_5
+    IL_00ff:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<int>.SetException(System.Exception)""
+    IL_0104:  nop
+    IL_0105:  leave.s    IL_011c
+  }
+ -IL_0107:  ldarg.0
+  IL_0108:  ldc.i4.s   -2
+  IL_010a:  stfld      ""int Test.<F>d__2.<>1__state""
+ ~IL_010f:  ldarg.0
+  IL_0110:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder<int> Test.<F>d__2.<>t__builder""
+  IL_0115:  ldloc.1
+  IL_0116:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<int>.SetResult(int)""
+  IL_011b:  nop
+  IL_011c:  ret
+}", sequencePoints: "Test+<F>d__2.MoveNext");
+        }
+
+        [Fact]
+        public void SpillSequencesRelease()
```

Comment:
One Debug and another Release.

---
In reply to: [127338049](https://github.com/dotnet/roslyn/pull/20772#discussion_r127338049) [](ancestors = 127338049)